frontend/packages/knative-plugin/src/components/add/event-sources/ContainerSourceSection.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useFormikContext, FormikValues } from 'formik';
+import { TextInputTypes, FormGroup } from '@patternfly/react-core';
+import { InputField, MultiColumnField } from '@console/shared';
+import { AsyncComponent } from '@console/internal/components/utils';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import { getSuggestedName } from '@console/dev-console/src/utils/imagestream-utils';
+
+const ContainerSourceSection: React.FC = () => {
+  const envPath = 'data.containersource.containers[0].env';
+  const { values, setFieldValue } = useFormikContext<FormikValues>();
+
+  const initialEnvValues = !_.isEmpty(values[envPath])
+    ? _.map(values[envPath], (env) => _.values(env))
+    : [['', '']];
+  const [nameValue, setNameValue] = React.useState(initialEnvValues);
+  const handleNameValuePairs = React.useCallback(
+    ({ nameValuePairs }) => {
+      const updatedNameValuePairs = _.compact(
+        nameValuePairs.map(([name, value]) => (value.length ? { name, value } : null)),
+      );
+      setNameValue(nameValuePairs);
+      setFieldValue(envPath, updatedNameValuePairs);
+    },
+    [setFieldValue],
+  );
+  return (
+    <FormSection title="ContainerSource">
+      <h3 className="co-section-heading-tertiary">Container</h3>
+      <InputField
+        data-test-id="container-image-field"
+        type={TextInputTypes.text}
+        name="data.containersource.containers[0].image"
```

Comment:
How about creating an enum/const  for these object path? Then you can use the same in setFieldValue also.

Reply:
created the path variable to hold all the paths.